lib/utils/get-theme.js

```diff
@@ -0,0 +1,12 @@
+export default (themeSetting, isElectron) => {
+  if (themeSetting === 'system') {
+    if (isElectron) {
+      const { remote } = __non_webpack_require__('electron'); // eslint-disable-line no-undef
+      return remote.systemPreferences.isDarkMode ? 'dark' : 'light';
+    }
+    return window.matchMedia('(prefers-color-scheme: dark)').matches
+      ? 'dark'
+      : 'light';
+  }
+  return themeSetting;
+};
```

Comment:
`isElectron` is something that won't change/can't change during runtime. because of that we won't need to pass it in to the function call on every invocation.

maybe it's easier to pass it in from `app.jsx` where we make the check, though it seems awkward.

it's probably also awkward to think about defining this function in `app.jsx`, but it would remove the arg-passing and place this directly next to where it's called (or in the same file at least vs. through a redirect)

okay so this is bad but here's an idea. it hooks into the point where we are supposedly initializing Electron specifics and replaces the `getTheme` function; by keeping it contained in `app.jsx` it removes some abstraction that maybe doesn't need to be there. it's less testable, but we're already depending on `remote.systemPreferences` and `window.matchMedia` which are themselves difficult to test

```js
class App extends Component {

  initializeElectron = () => {
    const { remote } = __non_webpack_require('electron');

    // replace theme selection to get system setting
    this.getTheme = () => {
      const { settings: { theme } } = this.props;

      return 'system' === theme
        ? ( remote.systemPreferences.isDarkMode ? 'dark' : 'light' ) 
        : theme;
    }

    this.setState( … );
  }

  …

  getTheme = () => {
    const { settings: { theme } } = this.props;

    return 'system' === theme
      ? ( window.matchMedia( '(prefers-color-scheme: dark)' ).matches ? 'dark' : 'light' )
      : theme;
  }

  …
}
```

broken apart…

```js
getSystemColorMode = () => window.matchMedia('(prefers-color-scheme: dark)').match ? 'dark' : 'light';

getTheme = () => {
  const { settings: { theme } } = this.props;

  return 'system' === theme
    ? this.getSystemColorMode()
    : theme;
}

initializeElectron = () => {
  …

  this.getSystemColorMode = () => remote.systemPreferences.isDarkMode ? 'dark' : 'light';

  …
}
```